From 82499d25e783a0914b1d2ec66682eb07a1212a0e Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sun, 06 Oct 2024 11:01:40 +0300
Subject: [PATCH] Revert "[SW-203304] package: fix build of agent-hbm"

This reverts commit 43df64411d9ed0fcca5769bcc0d87c4384d5aa9e.

The problem was found, no need for that

Change-Id: I9c3e04d0673f3e2e35c721d2f77bc03683a94b4b
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/board/habanalabs/fs1_arc/fs1-local.mk b/board/habanalabs/fs1_arc/fs1-local.mk
index 005d03f..aae1c14 100644
--- a/board/habanalabs/fs1_arc/fs1-local.mk
+++ b/board/habanalabs/fs1_arc/fs1-local.mk
@@ -11,6 +11,7 @@
 PREBOOT_D1_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-apps/preboot/app/preboot_d1
 AGENT_SERDES_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-plugins
 AGENT_PCIE_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-plugins
+AGENT_HBM_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-plugins
 DSM_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-apps/pcie-ss/dsm
 RESET_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-apps/reset/app/io
 RESET_COMPUTE_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-apps/reset/app/compute
diff --git a/configs/gaudi2_arm_defconfig b/configs/gaudi2_arm_defconfig
index 59c7e85..102a487 100644
--- a/configs/gaudi2_arm_defconfig
+++ b/configs/gaudi2_arm_defconfig
@@ -65,8 +65,8 @@
 BR2_PACKAGE_EEPROM_UTIL=y
 BR2_POST_IMAGE_UBOOT_ITS_TEMPLATE="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/uboot_extended.its"
 BR2_PACKAGE_AGENT_PCIE=y
-BR2_PACKAGE_ZEPHYR_AGENT_HBM=y
 BR2_PACKAGE_AGENT_SERDES=y
+BR2_PACKAGE_AGENT_HBM=y
 BR2_PACKAGE_AGENT_FWL=y
 BR2_PIC_PIE=n
 BR2_SSP_NONE=y
diff --git a/configs/gaudi3_arc_defconfig b/configs/gaudi3_arc_defconfig
index 108d7e4..62d711e 100644
--- a/configs/gaudi3_arc_defconfig
+++ b/configs/gaudi3_arc_defconfig
@@ -35,6 +35,6 @@
 BR2_RELRO_NONE=y
 BR2_FORTIFY_SOURCE_NONE=y
 BR2_PACKAGE_PALAD_BLOADER=y
-BR2_PACKAGE_ZEPHYR_AGENT_HBM=n
+BR2_PACKAGE_AGENT_HBM=n
 BR2_PACKAGE_AGENT_PCIE=y
 BR2_PACKAGE_AGENT_SERDES=y
diff --git a/package/agent-hbm/agent-hbm.mk b/package/agent-hbm/agent-hbm.mk
index 460aa61..6ee52f8 100644
--- a/package/agent-hbm/agent-hbm.mk
+++ b/package/agent-hbm/agent-hbm.mk
@@ -23,6 +23,19 @@
 		-I$(AGENT_SRC_DIR)/common/include \
 		-I$(AGENT_SRC_DIR)/common/include/$(HL_DEV)
 
+AGENT_HBM_CONF_APP_CFLAGS +=\
+		-I$(SPECS_EMB_PROJ_DIR) \
+		-I$(SPECS_EMB_PROJ_DIR)/$(DEV_PREFIX) \
+		-I$(SPECS_EMB_PROJ_DIR)/common \
+		-I$(SPECS_EMBEDDED_ROOT) \
+		-I$(DEVICE_SPECS_ROOT) \
+		-I$(DEVICE_SPECS_DIR) \
+		-I$(DEVICE_SPECS_DIR)/asic_reg \
+		-I$(SPECS_ROOT) \
+		-I$(SPECS_ROOT)/$(DEV_PREFIX) \
+		-I$(SPECS_ROOT)/$(DEV_PREFIX)/asic_reg \
+		-I$(SPECS_EXT_ROOT)
+
 AGENT_HBM_CONF_OPTS = $(ZEPHYR_CONF_OPTS)
 
 ifdef BR2_arcle
@@ -30,7 +43,7 @@
 endif
 ifdef BR2_aarch64
 AGENT_HBM_CONF_APP_EXTRA_CFLAGS += -mstrict-align
-AGENT_HBM_CONF_OPTS += -DBOARD=hl_$(AGENT_HBM_DEVICE_NAME)_$(HL_TARGET)
+AGENT_HBM_CONF_OPTS += -DBOARD=$(ZEPHYR_BOARD_PREFIX)_$(HL_TARGET)
 endif
 
 AGENT_HBM_BUILD_DIR = $(@D)
@@ -51,14 +64,9 @@
 AGENT_HBM_CONF_OPTS += -DFAKE_SECURITY=$(FAKE_SECURITY)
 AGENT_HBM_CONF_OPTS += -DBUILD_VERSION="$(AGENT_HBM_APP_VERSION)"
 AGENT_HBM_CONF_OPTS += -DBUILD_MACHINE="$(shell hostname -s)"
-AGENT_HBM_CONF_OPTS += -DCMAKE_C_FLAGS="\
-		-I$(SPECS_EMBEDDED_DIR) \
-		-I$(SPECS_EMBEDDED_DIR)/$(DEV_PREFIX) \
-		-I$(SPECS_ROOT) \
-		-I$(SPECS_EXT_ROOT) \
-		-I$(SPECS_ROOT)/$(DEV_PREFIX)/asic_reg \
-		$(AGENT_HBM_CONF_APP_CFLAGS)"
+AGENT_HBM_CONF_OPTS += -DCMAKE_C_FLAGS="$(AGENT_HBM_CONF_APP_CFLAGS)"
 
+AGENT_HBM_CONF_OPTS += -DDTS_EXTRA_CPPFLAGS="-I$(SPECS_EMB_PROJ_DIR)"
 AGENT_HBM_CONF_OPTS += -DDEVICE=$(AGENT_HBM_DEVICE_NAME)
 AGENT_HBM_CONF_OPTS += -DAPP_NAME=$(AGENT_HBM_APP_PROD_NAME)
 AGENT_HBM_CONF_OPTS += -DAPP_VERSION=$(AGENT_HBM_APP_VERSION)
diff --git a/package/zephyr/Config.in b/package/zephyr/Config.in
index 141d442..f7d036b 100644
--- a/package/zephyr/Config.in
+++ b/package/zephyr/Config.in
@@ -2,15 +2,3 @@
 	bool
 	help
 	  RTOS for ARC/ARM
-
-config BR2_PACKAGE_ZEPHYR_AGENT_HBM
-        bool "Zephyr agent hbm"
-        select BR2_PACKAGE_ZEPHYR
-        help
-          HBM on Zephyr
-
-config BR2_PACKAGE_ZEPHYR_AGENT_HBM
-        bool "Zephyr agent hbm"
-        select BR2_PACKAGE_ZEPHYR
-        help
-          HBM on Zephyr
diff --git a/package/zephyr/zephyr.mk b/package/zephyr/zephyr.mk
index 6abe77c..9b83ba0 100644
--- a/package/zephyr/zephyr.mk
+++ b/package/zephyr/zephyr.mk
@@ -130,10 +130,8 @@
 ZEPHYR_COMPONENTS_LIST += agent-fwl
 endif
 
-ifdef BR2_PACKAGE_ZEPHYR_AGENT_HBM
-ZEPHYR_HBM_MODULES = eeprom mbedtls boot_info i2c-indirect soc_init hl_shell mctp pldm
-$(call CREATE_ZEPHYR_MARGIN_TOOL_AGENT,hbm, $(HBM_EXTRA_CONF))
-ZEPHYR_CONFIGURE_CMDS += $(call CREATE_APP_VERSION,hbm,build-agent-hbm)
+ifdef BR2_PACKAGE_AGENT_HBM
+ZEPHYR_COMPONENTS_LIST += agent-hbm
 endif
 
 ifdef BR2_PACKAGE_AGENT_SERDES
